.github/workflows/build.yml

```diff
@@ -60,6 +60,49 @@ jobs:
         mv dist/psutil*.tar.gz wheelhouse/
         python scripts/internal/print_hashes.py wheelhouse/
 
+  # Windows cp37+ tests
+  # psutil tests do not like running from a virtualenv with python>=3.7 so
+  # not using cibuildwheel for those. run them "manually" with this job.
+  windows-py3-test:
+    name: windows-py3-test-${{ matrix.python }}-${{ matrix.architecture }}
+    needs: py3
+    runs-on: windows-2019
+    timeout-minutes: 20
+    strategy:
+      fail-fast: false
+      matrix:
+        python: ["3.7", "3.8", "3.9", "3.10", "~3.11.0-0"]
+        architecture: ["x86", "x64"]
+
+    steps:
+    - name: Cancel previous runs
+      uses: styfle/cancel-workflow-action@0.9.1
+      with:
+        access_token: ${{ github.token }}
+    - uses: actions/checkout@v3
+    - uses: actions/setup-python@v4
+      with:
+        python-version: "${{ matrix.python }}"
+        architecture: "${{ matrix.architecture }}"
+        cache: pip
+        cache-dependency-path: .github/workflows/build.yml
+    - name: Download wheels
+      uses: actions/download-artifact@v3
+      with:
+        name: wheels
+        path: wheelhouse
+    - name: Run tests
+      run: |
+        mkdir .tests
+        cd .tests
+        pip install $(find ../wheelhouse -name '*-cp36-abi3-${{ matrix.architecture == 'x86' && 'win32' || 'win_amd64'}}.whl')[test]
+        export PYTHONWARNINGS=always
+        export PYTHONUNBUFFERED=1
+        export PSUTIL_DEBUG=1
+        python ../psutil/tests/runner.py
+        python ../psutil/tests/test_memleaks.py
+      shell: bash
```

Comment:
Before I was against stop using appveyor, but with pyproject.toml, wheels and now `Py_LIMITED_API` I see why this is necessary. One question though: with this new addition GitHub will test setup.py using `Py_LIMITED_API`, while appveyor CI will test python 2.7 + setup.py **without** `Py_LIMITED_API`. Am I correct?  Do you still see value in keep using appveyor or do you think we should abandon it? 

Some background on why we're using appveyor:

* historical reason: initially (years ago) GitHub did not support windows
* to split work on 2 CI services and therefore complete the whole CI test/build sooner (but perhaps I'm wrong on this)

Reply:
I personally find using 2 CI services a bit confusing, and GHA's better integration with the repo/PRs is a nice advantage.

I believe there's no speed benefit to this split. GHA's concurrency limits are (much?) higher than the current jobs amount.

Reply:
> Am I correct? 

Yes you are.

> Do you still see value in keep using appveyor or do you think we should abandon it?

If you want to keep Python 2.7 wheels on Windows, then I'd say it shall remain. AppVeyor still keeps old images around  and that includes some build tools "[required](https://cibuildwheel.readthedocs.io/en/1.x/faq/#windows-and-python-27)" for Python 2.7 that are not available for download anymore and not available on GHA.

Given AppVeyor has 1 concurrent job and GHA has higher concurrency limits than the current jobs amount, it will be faster with GHA. The current limiter on `master` to get a green CI is AppVeyor with an average of 30 minutes. This PR cuts that at least in half.